src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Persistence/Extensions/AuditableEntityExtensions.cs

```diff
@@ -0,0 +1,56 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using Smart.FA.Catalog.Core.SeedWork;
+
+namespace Smart.FA.Catalog.Infrastructure.Persistence.Extensions;
+
+public static class EntryEntityAuditableDateUpdate
+{
+    /// <summary>
+    /// Updates a collection of <see cref="Entity" /> to keep their data auditable.
+    /// </summary>
+    /// <param name="entityEntries">A collection of <see cref="EntityEntry{TEntity}"/>.</param>
+    /// <param name="userId">The id of the user doing the current operation.</param>
+    public static void UpdateAutitableEntitiesData(this IEnumerable<EntityEntry<Entity>> entityEntries, int userId)
+    {
+        foreach (var entityEntry in entityEntries)
+        {
+            entityEntry.UpdateEntityAuditableData(userId);
+        }
+    }
+
+    private static void UpdateEntityAuditableData(this EntityEntry<Entity> entityEntry, int userId)
```

Comment:
To keep a coherent naming:
```suggestion
    private static void UpdateAuditableEntityData(this EntityEntry<Entity> entityEntry, int userId)
```